src/docfx/build/link/FileLinkMapBuilder.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+
+namespace Microsoft.Docs.Build
+{
+    internal class FileLinkMapBuilder
+    {
+        private readonly MonikerProvider _monikerProvider;
+        private readonly ListBuilder<FileLinkItem> _links = new ListBuilder<FileLinkItem>();
+
+        public FileLinkMapBuilder(MonikerProvider monikerProvider)
+        {
+            Debug.Assert(monikerProvider != null);
+            _monikerProvider = monikerProvider;
+        }
+
+        public void AddFileLink(Document file, string targetUrl)
+        {
+            Debug.Assert(file != null);
+            Debug.Assert(targetUrl != null);
+
+            var (_, monikers) = _monikerProvider.GetFileLevelMonikers(file);
```

Comment:
This line should have throw for TOC files due to https://github.com/dotnet/docfx/pull/5101 @928PJY

Reply:
No, that PR has been abandoned, and we decide to use the monkers from config and file metadata as the monikers of toc

---
In reply to: [327410727](https://github.com/dotnet/docfx/pull/5124#discussion_r327410727) [](ancestors = 327410727)

Reply:
Adding tests to cover this scenario as well